components/Heading/Heading.test.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Heading from './Heading';
+
+describe('Heading component', () => {
+  test('renders the correct heading level with default props', () => {
+    render(<Heading>Test Heading</Heading>);
+    const heading = screen.getByRole('heading');
+    expect(heading.tagName.toLowerCase()).toBe('h1');
+  });
+
+  test('applies the correct CSS classes based on the heading level', () => {
+    render(<Heading heading="h3">Test Heading</Heading>);
```

Comment:
Just like Fluent UI, use the attribute name `as` instea of heading.

Reply:
Added in as `as` attribute.